src/runtime/hexagon/hexagon_device_api.cc

```diff
@@ -167,10 +149,7 @@ void* HexagonDeviceAPI::AllocWorkspace(Device dev, size_t size, DLDataType type_
 
 void HexagonDeviceAPI::FreeWorkspace(Device dev, void* data) {
   CHECK(IsValidDevice(dev)) << "dev.device_type: " << dev.device_type;
-  CHECK(runtime_hexbuffs) << "Attempted to free Hexagon workspace with "
-                          << "HexagonDeviceAPI::FreeWorkspace outside of a session.  "
-                          << "Please call HexagonDeviceAPI::AcquireResources";
-  CHECK(runtime_hexbuffs->count(data) != 0)
+  CHECK(runtime_hexbuffs.count(data) != 0)
```

Comment:
This has the side effect that it will fail for things that were released in Release().  It isn't as clear that's the case with this refactor.

Reply:
I think this check is redundant to what's already done in the `FreeHexagonBuffer` method.